### PR TITLE
Create dedicated environment YML for Windows & reenable tests

### PR DESCRIPTION
**Pull request by michaelosthege**
This is based on a cherry-pick of the commit from #4779 which fixes the Windows step in the CI by using a more robust conda environment file.

I also re-enabled the Windows job in the CI, but did not yet add more test files.

This is the first step towards #4517

### CODE REVIEW

File: scripts/generate_pip_deps_from_conda.py [hunk 1]
```diff
@@ -117,7 +127,7 @@ def main(conda_fname, pip_fname):
         f"# This file is auto-generated by scripts/generate_pip_deps_from_conda.py, "
         "do not modify.\n# See that file for comments about the need/usage of each dependency.\n\n"
     )
-    pip_content = header + "\n".join(pip_deps) + "\n"
+    pip_content = header + "\n".join(sorted(pip_deps)) + "\n"
```
Comment (canyon289) [inline scripts/generate_pip_deps_from_conda.py:130]: Why does it have to be sorted?

File: conda-envs/windows-environment-dev-py38.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+name: pymc3-dev-py38
```
Comment (canyon289) [inline conda-envs/windows-environment-dev-py38.yml:1]: Is there any particular reason the window dev environment needs to be different? If so add a comment up top so whoever reads this next is aware what those are. 

I havet used windows in many years now so even I dont know, which is where the comment will help

**Review by canyon289 — APPROVED** [review]

File: scripts/generate_pip_deps_from_conda.py (hunk 1 above)
Comment (michaelosthege) [reply scripts/generate_pip_deps_from_conda.py:130]: In the Windows env file I split the dependencies by absolutely necessary vs. dev/docs related. The function here then took them in the same order, but `requirements-dev.txt` is sorted by another step in the precommit..

File: conda-envs/windows-environment-dev-py38.yml (hunk 2 above)
Comment (michaelosthege) [reply conda-envs/windows-environment-dev-py38.yml:1]: The `m2w64-toolchain` dependency is Windows-only, but conda doesn't support OS-conditional dependencies in these files..

I figured it was easier to maintain & diagnose two separate files than to find the set of dependencies that magically makes the conda dependency resolver find the correct result on all platforms.

Comment (canyon289) [reply conda-envs/windows-environment-dev-py38.yml:1]: I would add this explanation to the file

Comment (michaelosthege) [reply conda-envs/windows-environment-dev-py38.yml:1]: Deferred to #4786

**Pull request merged**